ARCH.md

```diff
@@ -0,0 +1,49 @@
+Rust-Lightning is broken into a number of high-level structures with various APIs to hook
+them together, as well as a number of APIs for you to provide external data which you,
+the user, have to provide.
+
+The two most important structures which nearly every application of Rust-Lightning will
+need to use are ChannelManager and ChannelMonitor. ChannelManager holds multiple Channels,
+routes payments between them and exposes a simple API to make and receive payments.
+Individual ChannelMonitors monitor the on-chain state of a Channel and punish
+counterparties if they misbehave, with the ManyChannelMonitor API providing a way for you
```

Comment:
"and force-close channel to time-out HTLC if required by timelock soon expiration"